python/pyspark/sql/connect/dataframe.py

```diff
@@ -822,6 +821,83 @@ def schema(self) -> StructType:
         else:
             return self._schema
 
+    @property
+    def isLocal(self) -> bool:
+        """Returns ``True`` if the :func:`collect` and :func:`take` methods can be run locally
+        (without any Spark executors).
+
+        .. versionadded:: 3.4.0
+
+        Returns
+        -------
+        bool
+        """
+        if self._plan is None:
+            raise Exception("Cannot analyze on empty plan.")
+        query = self._plan.to_proto(self._session)
+        return self._session._analyze(query).is_local
```

Comment:
are we going to cache the analyze result later?

Reply:
https://github.com/apache/spark/pull/38742#discussion_r1028915567

I think we will do the caching in near future.

Reply:
We literally can cache everything for each DataFrame since it is immutable. But I guess we need a design/discussion to clarify details of how.

Reply:
There is another interesting question is if we want to do caching on the server side.